#! /bin/sh /usr/share/dpatch/dpatch-run
## 40-fix-port-number-bigger-than-9.dpatch by Mario Izquierdo (mariodebian) <mariodebian@gmail.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Don't allow ports bigger than 9

@DPATCH@
diff -urNad p910nd~/p910nd.c p910nd/p910nd.c
--- p910nd~/p910nd.c	2009-02-16 11:02:25.000000000 +0100
+++ p910nd/p910nd.c	2009-02-16 11:06:50.000000000 +0100
@@ -589,6 +589,12 @@
 	argc -= optind;
 	argv += optind;
 	if (argc > 0) {
+		/* exit if port number is bigger than 9 */
+		/*(2 digits breaks some things, ex LOCKFILE) */
+		if ( strlen(argv[0]) > 1) {
+			fprintf(stderr, "Error, port number > 9\n");
+			return(-1);
+		}
 		if (isdigit(argv[0][0]))
 			lpnumber = argv[0][0];
 	}
